app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val _selectedPopularCategoryId: MutableStateFlow<Int> = MutableStateFlow(MovieCategory.POPULAR_STREAMING.ordinal)
+
+    val popularMoviesViewState: StateFlow<HomeMovieCategoryViewState> = _selectedPopularCategoryId.flatMapLatest {
+        movieRepository.popularMovies(MovieCategory.values()[it])
+            .map { movies ->
+                homeScreenMapper.toHomeMovieCategoryViewState(
+                    listOf(
+                        MovieCategory.POPULAR_STREAMING,
+                        MovieCategory.POPULAR_ON_TV,
+                        MovieCategory.POPULAR_FOR_RENT,
+                        MovieCategory.POPULAR_IN_THEATRES
+                    ),
+                    MovieCategory.values()[it], movies
+                )
+            }
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(1000L), HomeMovieCategoryViewState(emptyList(), emptyList()))
+
+    private val _selectedNowPlayingPopularCategoryId: MutableStateFlow<Int> = MutableStateFlow(MovieCategory.NOW_PLAYING_MOVIES.ordinal)
+
+    val nowPlayingMoviesViewState: StateFlow<HomeMovieCategoryViewState> = _selectedNowPlayingPopularCategoryId.flatMapLatest {
+        movieRepository.popularMovies(MovieCategory.values()[it])
+            .map { movies ->
```

Comment:
This is ok but it is easier for `_selectedNowPlayingPopularCategoryId` to be flow of `MovieCategory` instead of its ordinal. That way in flatMapLatest you can just pass it instead of searching for it using ordinal.